### ISSUE TITLE
Cache without APCu puts it in system temp dir instead of Symfony's cache dir

### ISSUE THREAD

**Issue by MattBred** [thecodingmachine/graphqlite-bundle#74]
I just encountered a problem where I had two different repositories utilising similar types cause a weird error - the applciation crashed (even just `bin/console`) because it couldn't find a service that doesn't exist. That service existed in the other repository.

No matter how many times I cleared the Symfony cache in `var/cache` and searched for this missing service in the second repository, I couldn't figure it out.

I finally stepped through the code in xdebug and found that `GraphqliteCompilerPass` function `getPsr16Cache()` has this tidbit

```php
            if (function_exists('apcu_fetch')) {
                $this->cache = new Psr16Cache(new ApcuAdapter('graphqlite_bundle'));
            } else {
                $this->cache = new Psr16Cache(new PhpFilesAdapter('graphqlite_bundle'));
            }
```

If the APCu extension isn't setup, it uses the `PhpFilesAdapter` which uses the `FilesystemCommonTrait`, that sets the cache dir with

```php
        if (!isset($directory[0])) {
            $directory = sys_get_temp_dir().\DIRECTORY_SEPARATOR.'symfony-cache';
```

Tl;DR: The cache gets set to `/tmp/symfony-cache` which is not configurable from what I can tell. This bundle should be setting the cache to the default Symfony cache directory.

Is there something I'm missing?

**Comment by MattBred** [comment]
This would be a really simple fix I think?

```diff
diff --git a/DependencyInjection/GraphqliteCompilerPass.php b/DependencyInjection/GraphqliteCompilerPass.php
index 3d03c26..261b199 100644
--- a/DependencyInjection/GraphqliteCompilerPass.php
+++ b/DependencyInjection/GraphqliteCompilerPass.php
@@ -91,12 +91,18 @@ class GraphqliteCompilerPass implements CompilerPassInterface
      */
     private $annotationReader;
 
+    /**
+     * @var string
+     */
+    private $cacheDir;
+
     /**
      * You can modify the container here before it is dumped to PHP code.
      */
     public function process(ContainerBuilder $container): void
     {
         $reader = $this->getAnnotationReader();
+        $cacheDir = $container->getParameter('kernel.cache_dir');
         //$inputTypeUtils = new InputTypeUtils($reader, $namingStrategy);
 
         // Let's scan the whole container and tag the services that belong to the namespace we want to inspect.
@@ -442,7 +448,7 @@ class GraphqliteCompilerPass implements CompilerPassInterface
             if (function_exists('apcu_fetch')) {
                 $this->cache = new Psr16Cache(new ApcuAdapter('graphqlite_bundle'));
             } else {
-                $this->cache = new Psr16Cache(new PhpFilesAdapter('graphqlite_bundle'));
+                $this->cache = new Psr16Cache(new PhpFilesAdapter('graphqlite_bundle', 0, $this->cacheDir));
             }
         }
         return $this->cache;
```

**Comment by PhilippSchreiber** [comment]
I can confirm that this causes trouble. 

Especially when deploying with another user then the web user. When I run cache:warmup the www-data user cannot read the information afterwards. For the regular symfony cache folder this is fixed with an ACL but for folders with dynamic names it is not easy to fix.

**Comment by PhilippSchreiber** [comment]
Unfortunately the fix does not work (even with accessing the right variable). This also causes problems when running functional tests as the cache gets warmed up by the (in our case) vagrant user und then the web user is not allowed to access the files.

**Comment by PhilippSchreiber** [comment]
I solved this by configuring the php temp folder to a fixed folder where I can use ACL.

**Comment by MattBred** [comment]
I don't think configuring the php temp folder is the right play here. This is the Symfony bundle, so IMO it should use the Symfony cache.

What was the issue with my solution above? If it's just permission problems, that can be solved after the cache warmup and before the tests?

**Comment by PhilippSchreiber** [comment]
The problem is, that the underlaying thecodingmachine/graphqlite package uses the php temp dir together with a generated number from src path. And in the systems tmp folder you cannot use an ACL. This is the reason, why we configured the php tmp folder to a location where we can use ACL.

**Comment by PhilippSchreiber** [comment]
So perhaps you are right and you fix works for this specific problem (with modifications) and I should open another issue at thecodingmachine/graphqlite

**Comment by MattBred** [comment]
Making a PR
